src/basedtyping/transformer.py

```diff
@@ -0,0 +1,253 @@
+"""utilities to create standard compatible annotations"""
+
+from __future__ import annotations
+
+import ast
+import types
+import typing
+import uuid
+from contextlib import contextmanager
+from enum import Enum
+from typing import cast
+
+import typing_extensions
+from typing_extensions import override
+
+import basedtyping
+
+
+# ruff: noqa: S101
+class CringeTransformer(ast.NodeTransformer):
+    """Transforms `1 | 2` into `Literal[1] | Literal[2]` etc"""
+
+    def __init__(
+        self,
+        globalns: dict[str, object] | None,
+        localns: dict[str, object] | None,
+        *,
+        string_literals: bool,
+    ):
+        self.string_literals = string_literals
+
+        # This logic for handling Nones is copied from typing.ForwardRef._evaluate
+        if globalns is None and localns is None:
+            globalns = localns = {}
+        elif globalns is None:
+            assert localns is not None
+            globalns = localns
+        elif localns is None:
+            assert globalns is not None
+            localns = globalns
+
+        self.typing_name = f"_typing_extensions_{uuid.uuid4().hex}"
+        self.basedtyping_name = f"_basedtyping_{uuid.uuid4().hex}"
+        self.globalns = globalns
+
+        self.localns = {
+            **localns,
+            self.typing_name: typing_extensions,
+            self.basedtyping_name: basedtyping,
+        }
```

Comment:
because `python>=3.9` now, you could

```suggestion
        self.localns = localns | {
            self.typing_name: typing_extensions,
            self.basedtyping_name: basedtyping,
        }
```

Reply:
okay, just don't tell detach or well have to scrap the entire project

Reply:
oh no, he found it